### PR TITLE
GitHub Actions: Combine mypy and pytest jobs

### LINKED ISSUE #37: Merge pytest and mypy into a single run
There seems to be a lot of duplicate setup and teardown in the current GitHub Actions...

Why not save time and resources by doing mypy and then pytest (or vice versa) in the same job?

### PR DESCRIPTION
**Pull request by cclauss**
Fixes #37 
* #37 

Adds pylint and other ruff rules.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# This workflow will install Python dependencies, run tests and lint with a
-# variety of Python versions
+# This workflow will install Python dependencies, do type checking, and
+# run tests with variety of Python versions.
```
Comment (cclauss) [inline .github/workflows/check.yml:2]:
```suggestion
# run tests with a variety of Python versions.
```

File: nntp/yenc.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def decode(self, buf: bytes) -> bytes:
             elif b == 0x3D:
                 self._escape = 1
                 continue
-            elif b == 0x0D or b == 0x0A:
+            elif b in (13, 10):
```
Comment (cclauss) [inline nntp/yenc.py:60]:
```suggestion
            elif b in {0x0D, 0x0A}:
```

File: nntp/headerdict.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def __init__(
     def __getitem__(self, key: str) -> str:
         return self.__proxy[HeaderName(key)]
 
-    def __setitem__(self, key: str, value: Any) -> None:
+    def __setitem__(self, key: str, value: Any) -> None:  # noqa: ANN401
```
Comment (greenbender) [inline nntp/headerdict.py:62]: `value` should be typed as `str` here anyway.

File: pyproject.toml [hunk 4]
```diff
@@ -72,33 +82,33 @@ select = [
...
 [tool.ruff.lint.mccabe]
 max-complexity = 12
 
+[tool.ruff.lint.pylint]
+allow-magic-value-types = ["bytes", "int", "str"]
+max-args = 8  # Default: 5
+
 [tool.ruff.lint.per-file-ignores]
+"nntp/nntp.py" = ["A002", "ARG002"]
+"nntp/utils.py" = ["N806"]
```
Comment (greenbender) [inline pyproject.toml:113]: How many instances of each of these are there? Is it worth fixing these instead? I'm happy to do this in a follow-up.

File: nntp/headerdict.py (hunk 3 above)
Comment (cclauss) [inline nntp/headerdict.py:62]:
```suggestion
    def __setitem__(self, key: str, value: str) -> None:
```

File: pyproject.toml (hunk 4 above)
Comment (greenbender) [reply pyproject.toml:113]: Thanks! I'll fix the ones that make sense in a follow up.

**Pull request merged**